sklearn/_config.py

```diff
@@ -197,6 +208,12 @@ def config_context(
 
         .. versionadded:: 1.1
 
+    array_api_dispatch : bool, default=None
```

Comment:
I think all the configuration options uses `None` as a default because we do not want `set_config()` to override the configuration when the configuration option is not passed in. For example:

```python
set_config(array_api_dispatch=True)

# We do not want this to override `array_api_dispatch=True`
set_config(assume_finite=True)
```

Reply:
Indeed.